osu.Game/Screens/Play/DimmableStoryboard.cs

```diff
@@ -24,6 +26,21 @@ public partial class DimmableStoryboard : UserDimContainer
         private readonly Storyboard storyboard;
         private readonly IReadOnlyList<Mod> mods;
 
+        /// <summary>
+        /// In certain circumstances, the storyboard cannot be hidden entirely even if it is fully dimmed. Such circumstances include:
+        /// <list type="bullet">
+        /// <item>
+        /// cases where the storyboard has an overlay layer sprite, as it should continue to display fully dimmed
+        /// <i>in front of</i> the playfield (https://github.com/ppy/osu/issues/29867),
+        /// </item>
+        /// <item>
+        /// cases where the storyboard includes samples - as they are played back via drawable samples,
+        /// they must be present for the playback to occur (https://github.com/ppy/osu/issues/9315).
+        /// </item>
+        /// </list>
+        /// </summary>
+        private readonly Lazy<bool> storyboardMustAlwaysBePresent;
```

Comment:
The `Lazy` usage here is primarily intended to prevent the recalculation of the heuristic conditions described, which _can_ get pretty heavy on maps like https://osu.ppy.sh/beatmapsets/470977#osu/1010030 (measured at least 3ms spent on computing this in profiling per call). That said, it doesn't do _that_ much, so it could be skipped if deemed offensive.

Reply:
Any reason for it to not be computed once in `LoadComplete()` instead?

Reply:
Um... I guess you could, but it gets rather finicky, because if you do that you need to start worrying about call order - `ShowDimContent` is checked in `UpdateVisuals()` in `UserDimContainer` which is called in several places there, and one of them is... `UserDimContainer.LoadComplete()`. So one would have to be careful to compute this before that base call.